build/record.go

```diff
@@ -155,3 +180,54 @@ func encodeProvenance(dt []byte, mode confutil.MetadataProvenanceMode) (string,
 	}
 	return base64.StdEncoding.EncodeToString(dtprv), nil
 }
+
+func setRecordStatus(ctx context.Context, c *client.Client, ref string, updateExporterResponse func(key string, value string), pw progress.Writer) error {
+	mode := confutil.MetadataStatus()
+	if mode == confutil.MetadataStatusModeDisabled {
+		return nil
+	}
+	pw = progress.ResetTime(pw)
+	return progress.Wrap("resolving status for metadata file", pw.Write, func(l progress.SubLogger) error {
```

Comment:
I don't think this should be pulled via history API and new progress item added. You can get the warnings by calling `printer.Warnings()` from command packages.

Reply:
Opened https://github.com/docker/buildx/pull/2551 as alternative